AlphaCentauri/Assets/Scripts/RodCasting.cs

```diff
@@ -28,13 +32,66 @@ void BobberThrow()
         float time = distance.magnitude / bobberVelocity;
         rigidbody.velocity = bobberVelocity * distance.normalized + new Vector3(0, time * Physics.gravity.magnitude * 0.5f);
         // bobberClone.position = target.position;
+        StartFishing();
+        //Trigger the fish eat the bait
     }
     void SetPowerLevel()
     {
         amplitude = horizontalBar.parent.GetComponent<RectTransform>().rect.width - horizontalBar.GetComponent<RectTransform>().rect.width;
         playHorizontal = true;
     }
     // Tween the bar of rightLeft or upDown
+    void StartFishing()
+    {
+        StartCoroutine(FishingCorountine());
+    }
+    IEnumerator FishingCorountine()
+    {
+        if (IsFishing)
+        {
+            int wait_time = Random.Range (5, 10);//Wait random time to fish to bite the bait
+		    yield return new WaitForSeconds (wait_time);
+        }
+        if (IsFishing)
+        {
+            Debug.Log("Biting");
+            StartCoroutine(StartFishStruggle());//Start fish struggle
+        }
+        
+    }
+    IEnumerator StartFishStruggle()
+    {
+        isThereABite = true;
+        //wait until pull
+        while (!Pulling)
+        {
+            yield return FishHadEnough +=1f;
+            if(FishHadEnough>= 3000)
```

Comment:
1. dont put arbitrary numbers here btw, use a variable and put it on the top of the file so that when play testing you can easily change it

2. this is 3000 frames, if someone runs on 30 fps it would take 100 seconds, if someone runs on 60 fps only 50 seconds. dont use this, use the waitforseconds so its consistent